### ISSUE TITLE
Share 3D models?

### ISSUE THREAD

**Issue by SYUMJOBA** [mastodon/mastodon#31332]
### Pitch

The idea is simple, over images add the possibility to share 3D models with an embedded visualizer, like in a three.js canvas, amongst other files, with `OrbitControls` enabled. [example](https://threejs.org/examples/#misc_controls_orbit)

Kind of like how sketchfab does it.

.GLTF, .GLB, .OBJ and .STL and maybe even .VOX (all are openly supported through THREE.JS) really is what I have in mind here.

### Motivation

I'm completely new to mastodon, discovered it through a reddit thread, but I figured it is a pretty beautiful social, specifically for the fact that is has the chance to properly be a sharing platform of passionate projects and interesting discussions :D

Many individuals would benefit from it, I myself would enjoy such a thing, artists would love this, especially in an era where there's AI flooding the internet.

**Comment by RickDW** [comment]
Welcome to Mastodon! I became a member on the mathstodon.xyz instance about two years ago. This instance's server has been modified to render mathematics using a Latex renderer in the web interface. Extending the usual text/image/video content that everyone uses is really nice.
It might be good to implement this on a specific instance where many users are interested in using this, at least as a first step I think. Maybe on a game development instance like https://mastodon.gamedev.place ?

**Comment by ThisIsMissEm** [comment]
There is also a fediverse service that is dedicated to 3D models, though I'm at a loss for its name right now.

I'd be inclined to say that native support in Mastodon core for 3D models is very unlikely.

**Comment by SYUMJOBA** [comment]
Yeah as I imagined, 3D models can be expensive to host to, this was just an idea that I would be very interested to see implement :D I think sketchfab links would do enough, maybe embedding them smartly when sketchfab link is published? That could be a quick and easy fix!

**Comment by ThisIsMissEm** [comment]
If they have good opengraph metadata, then perhaps something might be possible?
